File: workbench-server/src/main/java/org/example/workbenchserver/entity/Anniversary.java
package org.example.workbenchserver.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableLogic;
import com.baomidou.mybatisplus.annotation.TableName;

import java.time.LocalDateTime;

/**
 * 生日与纪念日实体，对应 {@code wb_anniversary}（docs/数据模型.md §4.3）。
 *
 * <p>生日和纪念日**共用这一张表**，靠 {@code type} 区分 —— 两者字段几乎一致，
 * 拆两张表纯属重复。
 *
 * <p><b>本类刻意没有 {@code userId} 字段，这不是漏写</b>，理由同
 * {@link PlanTask}：租户插件只在 INSERT 时补列清单里没有的列，
 * 实体一旦带 {@code userId} 就成了绕过隔离的越权写入通道。
 *
 * <p>只存 {@code month} + {@code day}、不存年份：生日每年重复，年份没有意义。
 * 代价是跨年推算得在 Java 里做，见 {@code YearlyRecurrence}。
 * 将来若要显示年龄或"第几年"，需要补 {@code year} 字段（docs/数据模型.md §4.3 已注明）。
 *
 * <p>{@code createTime} / {@code updateTime} 交给数据库的
 * {@code DEFAULT CURRENT_TIMESTAMP} 与 {@code ON UPDATE CURRENT_TIMESTAMP}。
 */
@TableName("wb_anniversary")
public class Anniversary {

	@TableId(type = IdType.AUTO)
	private Long id;

	/** 生日填姓名，纪念日填名称 */
	private String name;

	/** 1 生日 / 2 纪念日 */
	private Integer type;

	/** 关系：自己 / 家人 / 朋友。仅生日有意义，纪念日留空 */
	private String relation;

	/** 月 1-12 */
	private Integer month;

	/** 日 1-31。2 月允许存 29（闰年出生），平年由 YearlyRecurrence 退到 2/28 */
	private Integer day;

	/** 提前提醒天数 1-7 */
	private Integer remindDays;

	private String remark;

	private LocalDateTime createTime;

	private LocalDateTime updateTime;

	/** 逻辑删除：0 未删除，非 0 为删除时间戳 */
	@TableLogic
	private Long deleted;

	public Long getId() {
		return id;
	}

	public void setId(Long id) {
		this.id = id;
	}

	public String getName() {
		return name;
	}

	public void setName(String name) {
		this.name = name;
	}

	public Integer getType() {
		return type;
	}

	public void setType(Integer type) {
		this.type = type;
	}

	public String getRelation() {
		return relation;
	}

	public void setRelation(String relation) {
		this.relation = relation;
	}

	public Integer getMonth() {
		return month;
	}

	public void setMonth(Integer month) {
		this.month = month;
	}

	public Integer getDay() {
		return day;
	}

	public void setDay(Integer day) {
		this.day = day;
	}

	public Integer getRemindDays() {
		return remindDays;
	}

	public void setRemindDays(Integer remindDays) {
		this.remindDays = remindDays;
	}

	public String getRemark() {
		return remark;
	}

	public void setRemark(String remark) {
		this.remark = remark;
	}

	public LocalDateTime getCreateTime() {
		return createTime;
	}

	public void setCreateTime(LocalDateTime createTime) {
		this.createTime = createTime;
	}

	public LocalDateTime getUpdateTime() {
		return updateTime;
	}

	public void setUpdateTime(LocalDateTime updateTime) {
		this.updateTime = updateTime;
	}

	public Long getDeleted() {
		return deleted;
	}

	public void setDeleted(Long deleted) {
		this.deleted = deleted;
	}

}

File: stfworkbench-vue/src/types/anniversary.ts
/**
 * 生日与纪念日相关类型。字段与后端 AnniversaryVO / UpcomingAnniversaryVO
 * 一一对应，见 docs/接口清单.md §5。
 */

/** 类型：1 生日 / 2 纪念日 */
export const TYPE_BIRTHDAY = 1
export const TYPE_MEMORIAL = 2

/** 常用关系。不限制用户只能选这几个 —— 表单里允许自己输入 */
export const RELATION_OPTIONS = ['自己', '家人', '朋友']

export interface Anniversary {
  id: number
  name: string
  type: number
  relation: string
  /** 月 1-12 */
  month: number
  /** 日 1-31（2 月可存 29，闰年生的人也要录得进来） */
  day: number
  /** 提前提醒天数 1-7 */
  remindDays: number
  remark: string
}

/**
 * 新增 / 修改的入参。
 *
 * 两者共用一份：后端的 PUT 是全量替换（字段集合与 POST 完全一致），
 * 不像每日计划那样有"不给就不改"的字段，所以不需要拆成两个类型。
 */
export interface AnniversaryParams {
  name: string
  type: number
  relation: string
  month: number
  day: number
  remindDays: number
  remark: string
}

/**
 * 即将到来的一条，供提醒用。
 *
 * 只有这个结构带 `daysUntil` / `nextDate` —— 它们**由后端算好**，
 * 前端直接用，不要自己拿 month/day 去推（见 docs/接口清单.md §5）。
 * 生日只存月日的代价就是跨年推算，那件事在后端做了一处、测了一处，
 * 前端再算一遍就多出一个可能对不上的口径。
 */
export interface UpcomingAnniversary {
  id: number
  name: string
  type: number
  relation: string
  month: number
  day: number
  /** 距下次还有几天，当天为 0 */
  daysUntil: number
  /** 下次落在哪天，yyyy-MM-dd */
  nextDate: string
}
